src/lib/array.js

```diff
@@ -48,6 +51,135 @@ exports.ensureArray = function(out, n) {
     return out;
 };
 
+var typedArrays = {
+    c1: typeof Uint8ClampedArray === 'undefined' ? undefined :
+               Uint8ClampedArray, // not supported in numpy?
+
+    i1: typeof Int8Array === 'undefined' ? undefined :
+               Int8Array,
+
+    u1: typeof Uint8Array === 'undefined' ? undefined :
+               Uint8Array,
+
+    i2: typeof Int16Array === 'undefined' ? undefined :
+               Int16Array,
+
+    u2: typeof Uint16Array === 'undefined' ? undefined :
+               Uint16Array,
+
+    i4: typeof Int32Array === 'undefined' ? undefined :
+               Int32Array,
+
+    u4: typeof Uint32Array === 'undefined' ? undefined :
+               Uint32Array,
+
+    f4: typeof Float32Array === 'undefined' ? undefined :
+               Float32Array,
+
+    f8: typeof Float64Array === 'undefined' ? undefined :
+               Float64Array,
+
+    /* TODO: potentially add Big Int
+
+    i8: typeof BigInt64Array === 'undefined' ? undefined :
+               BigInt64Array,
+
+    u8: typeof BigUint64Array === 'undefined' ? undefined :
+               BigUint64Array,
```

Comment:
Turns out if you make a numpy array from a list of integers, even if they're all small, seems like `int64` is the default, but if a float is included then `float64` becomes the default.
```py
>>> import numpy as np
>>> a=np.array([1,2,3,4,5])
>>> a
array([1, 2, 3, 4, 5])
>>> a.dtype
dtype('int64')
>>> b=np.array([1.0,2])
>>> b.dtype
dtype('float64')
```
I don't know if Pandas does anything smarter than this, but I bet just from ^^ we're going to see a good deal of `int64` so probably a good idea to turn it on now.

Reply:
Good point. These cases would be handled in https://github.com/plotly/plotly.py/pull/4470/commits/86a6c55afd11d034e67386d8457b59de078cbfce

Reply:
And if int64 (Big Int) is really needed plotly.py layer pass those in regular arrays. So we don't hit a problem.